pkg/metrics/metrics.go

```diff
@@ -151,3 +193,37 @@ func (mc MachineCollector) listMachines() ([]*mapiv1beta1.Machine, error) {
 func (mc MachineCollector) listMachineSets() ([]*mapiv1beta1.MachineSet, error) {
 	return mc.machineSetLister.MachineSets(mc.namespace).List(labels.Everything())
 }
+
+const (
+	VSphereProvider string = "vsphere"
```

Comment:
why do we need this? The platform is implicit in any running cluster.

Reply:
I was assuming the failure rates will be collected across clusters, and aggregated by that value in a longer term?

Reply:
These metrics are only exposed at each individual cluster scope. If anything were to aggregate them in the future we will consider introduce any further suitable label by then.

Reply:
Would we not want to have this label if we were bring telemetry back to RH about our controllers?

Reply:
That's my idea too. May be a better idea to embed the provider name resolving into MAO, so it will identify it from infrastructure config and set it before sending metrics, @enxebre @JoelSpeed?

Reply:
>Would we not want to have this label if we were bring telemetry back to RH about our controllers?

If we do we, we can discuss it when there's a proposal for exposing machine API metrics for telemetry. As for today this property does not provide any value for the consumers of these metrics, e.g users owning the cluster.